addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -306,3 +306,58 @@ def test_recompute_partner_from_task_customer_change(self):
         self.task1.write({'partner_id': partner2})
 
         self.assertEqual(timesheet_entry.partner_id, partner2, "The timesheet entry's partner should still be equal to the task's partner/customer, after the change")
+
+    def test_task_with_timesheet_project_change(self):
+        project_manager = self.env['res.users'].create({
+            'name': 'user_project_manager',
+            'login': 'user_project_manager',
+            'groups_id': [(6, 0, [self.ref('project.group_project_manager')])],
+        })
+
+        second_project_manager = self.env['res.users'].create({
+            'name': 'user_second_project_manager',
+            'login': 'user_second_project_manager',
+            'groups_id': [(6, 0, [self.ref('project.group_project_manager')])],
+        })
```

Comment:
We don't need another project_manager in my opinion, we just want to test that a user having the group `project.group_project_manager` can write on a timesheet.
```suggestion
```